params/config.go

```diff
@@ -633,6 +633,7 @@ type ChainConfig struct {
 	CurieBlock          *big.Int `json:"curieBlock,omitempty"`          // Curie switch block (nil = no fork, 0 = already on curie)
 	DarwinTime          *uint64  `json:"darwinTime,omitempty"`          // Darwin switch time (nil = no fork, 0 = already on darwin)
 	DarwinV2Time        *uint64  `json:"darwinv2Time,omitempty"`        // DarwinV2 switch time (nil = no fork, 0 = already on darwinv2)
+	EuclidBlock         *big.Int `json:"euclidBlock,omitempty"`         // Euclid switch block (nil = no fork, 0 = already on euclid)
```

Comment:
Euclid will be triggered not by block but by time. You can probably just copy this function and the definition of `EuclidTime *uint64` from here: https://github.com/scroll-tech/go-ethereum/pull/1104/files#diff-10ff00a15ef58da7485fedeca3906c73cb236fde4b143a9d61601c63cbf1ffe8R899